hollow/src/main/java/com/netflix/hollow/tools/diff/HollowDiffMatcher.java

```diff
@@ -91,7 +95,14 @@ public void calculateMatches() {
         int candidateToMatchOrdinal = toPopulatedOrdinals.nextSetBit(0);
         while(candidateToMatchOrdinal != -1) {
             Object key[] = toIdx.getRecordKey(candidateToMatchOrdinal);
-            int matchedOrdinal = fromIdx.getMatchingOrdinal(key);
+            int matchedOrdinal = -1;
+            try {
+                matchedOrdinal = fromIdx.getMatchingOrdinal(key);
+            } catch(NullPointerException ex) {
+                LOGGER.error("Null values found for type " + toTypeState.getSchema().getName()
+                        + " with key field values " + Arrays.asList(key) + " at ordinal : " + candidateToMatchOrdinal
+                        + " with stacktrace : " + ex.getStackTrace());
```

Comment:
NOTE: double check that this is logging the stack trace.

Reply:
ack. have removed it from printing. As discussed, re throwing this as RuntimeException to preserve existing behavior for Hollow Tooling